pyccel/stdlib/pyc_math/pyc_math.c

```diff
@@ -0,0 +1,43 @@
+#include "pyc_math.h"
+#include <math.h>
+
+/*---------------------------------------------------------------------------*/
+long        pyc_factorial(long n)
+{
+    long    res = 1;
+
+    /* ValueError: factorial() not defined for negative values */
+    if (n < 0)
+        return 0;
+    for (int i = 2; i <= n; i++)
+        res *= i;
+    return (res);
+}
```

Comment:
Do we need multiple versions for different precisions? Perhaps the maximum precision is sufficient. Is `long` the largest integer handled by pyccel?

Reply:
the equivalent of the python integer in c is long. why we need other precision?
maybe we will need that for numpy factorial.

Reply:
Ah you are right, we can't pass numpy types to the math library :+1: